src/ORM/Table.php

```diff
@@ -872,14 +872,27 @@ public function hasBehavior($name)
     }
 
     /**
-     * Returns an association object configured for the specified alias if any
+     * Returns an association object configured for the specified alias if any.
+     *
+     * The name argument also supports dot syntax to access deeper associations.
+     *
+     * ```
+     * $users = $this->association('Articles.Comments.Users');
+     * ```
      *
      * @param string $name the alias used for the association.
      * @return \Cake\ORM\Association|null Either the association or null.
      */
     public function association($name)
     {
-        return $this->_associations->get($name);
+        list($name, $next) = array_pad(explode('.', $name, 2), 2, null);
```

Comment:
Optimization should be added for checking whether name contains `.` and return early from the method.